packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -163,6 +164,20 @@ abstract class Logger {
     bool? wrap,
   });
 
+  /// Display the [message] inside a box.
+  /// For example, this is the generated output:
+  ///
+  ///   ┌─ [title] ─┐
+  ///   │ [message] │
+  ///   └───────────┘
+  ///
+  /// If a terminal is attached, the lines in [message] are automatically wrapped based on
+  /// the available columns.
+  void printBox(
```

Comment:
we should document our UI convention around using this. Do we use it for every message? max one message per run? only for serious errors? only for informative messages? progress? etc.

Reply:
done

Reply:
as we develop our use of this we should continue to elaborate on our conventions here (e.g. is it ok to show two boxes in one run?) but this is a good start.